### ISSUE TITLE
REST API populate only works with drafts

### ISSUE THREAD

**Issue by Stefan-SW** [strapi/strapi#21004]
### Required System information

- Node.js version:  20.12.2
- Yarn version: 4.4.0
- Strapi version: 5.0.0-rc.9
- Database: SqLite
- Operating system: linux-x64 (Debian GNU/Linux 11 bullseye)
- Is your project Javascript or Typescript: typescript

### Describe the bug

When populating a oneToOne relation the REST API won't fill in the entity unless the `status` query parameter is set to `draft`. (When the parameter is set to `draft` the draft version of the related entity is returned.)

### Steps to reproduce the behavior

1.Create the following collection types:

` /api/principal/content-types/principal/schema.json`
```json
{
  "kind": "collectionType",
  "collectionName": "principals",
  "info": {
    "singularName": "principal",
    "pluralName": "principals",
    "displayName": "principal"
  },
  "options": {
    "draftAndPublish": true
  },
  "pluginOptions": {},
  "attributes": {
    "name": {
      "type": "string"
    }
  }
}
```

`/api/dependent/content-types/dependent/schema.json`
```json
{
  "kind": "collectionType",
  "collectionName": "dependents",
  "info": {
    "singularName": "dependent",
    "pluralName": "dependents",
    "displayName": "dependent"
  },
  "options": {
    "draftAndPublish": false
  },
  "pluginOptions": {},
  "attributes": {
    "Name": {
      "type": "string"
    },
    "principal": {
      "type": "relation",
      "relation": "oneToOne",
      "target": "api::principal.principal"
    }
  }
}
```

2. create an instance of the `principal` entity and publish it.
3. create an instance of the `dependent` entity, link the `pricipal` entity from step 2 in the `pricipal` field and save it.
4. Fetch `dependents` and populate the `pricipal` field:
`> http 'localhost:1337/api/dependents?populate=principal'`
```json
{
    "data": [
        {
            "Name": "the dependent entity",
            "createdAt": "2024-08-13T08:52:14.348Z",
            "documentId": "u48wv1ugkfwmj3dm41hzznz1",
            "id": 1,
            "locale": null,
            "principal": null,
            "publishedAt": "2024-08-13T08:52:14.340Z",
            "updatedAt": "2024-08-13T08:52:14.348Z"
        }
    ],
    "meta": {
        "pagination": {
            "page": 1,
            "pageCount": 1,
            "pageSize": 25,
            "total": 1
        }
    }
}
```
The response **does not** contain the *published version* of the `principal` entity.         
5. Fetch `dependents` and populate the `pricipal`  **and* set the `status` parameter to `draft`
`http 'localhost:1337/api/dependents?populate=principal&status=draft'`
```json
{
    "data": [
        {
            "Name": "the dependent entity",
            "createdAt": "2024-08-13T08:52:14.348Z",
            "documentId": "u48wv1ugkfwmj3dm41hzznz1",
            "id": 1,
            "locale": null,
            "principal": {
                "createdAt": "2024-08-13T08:51:58.788Z",
                "documentId": "knjgnlldejwhim7zqtv9wfu9",
                "id": 1,
                "locale": null,
                "name": "the principal entity",
                "publishedAt": null,
                "updatedAt": "2024-08-13T08:51:58.788Z"
            },
            "publishedAt": "2024-08-13T08:52:14.340Z",
            "updatedAt": "2024-08-13T08:52:14.348Z"
        }
    ],
    "meta": {
        "pagination": {
            "page": 1,
            "pageCount": 1,
            "pageSize": 25,
            "total": 1
        }
    }
}
```
This time the response **does contain** the *draft version* of the `principal` entity.

### Expected behavior

I expect the REST API to populate relations with the published version by default.

The query 
`> http 'localhost:1337/api/dependents?populate=principal'`
should return
```json
{
    "data": [
        {
            "Name": "the dependent entity",
            "createdAt": "2024-08-13T08:52:14.348Z",
            "documentId": "u48wv1ugkfwmj3dm41hzznz1",
            "id": 1,
            "locale": null,
            "principal": {
                "createdAt": "2024-08-13T08:51:58.788Z",
                "documentId": "knjgnlldejwhim7zqtv9wfu9",
                "id": 1,
                "locale": null,
                "name": "the principal entity",
                "publishedAt": "2024-08-13T08:52:14.340Z",
                "updatedAt": "2024-08-13T08:51:58.788Z"
            },
            "publishedAt": "2024-08-13T08:52:14.340Z",
            "updatedAt": "2024-08-13T08:52:14.348Z"
        }
    ],
    "meta": {
        "pagination": {
            "page": 1,
            "pageCount": 1,
            "pageSize": 25,
            "total": 1
        }
    }
}
```

**Comment by CodeToTee** [comment]
Adding some (hopefully) useful info to this issue.  I'm seeing the same problem (collections not populating), with the following observations:
- If a collection has a oneToOne relationship with itself, it will populate
- If a collection has a oneToOne with *another* collection, it will not populate
- If the relationship is 'has and belongs to' (i.e. oneToOne with a back-reference), it will populate

Both the above are true irrespective of the value of 'status'.  This is observed on rc.12.

I've enabled database debug, and I can see the SQL queries to the primary collection, then the SQL query to the secondary collection.  When I replay those SQL commands to the database, with the attached parameters, I get the data I expect.  However, that data isn't being populated in the API response.  This only seems to happen when these is no inverse relation (i.e. no reference from the secondary back to the primary).

Hopefully this will help pinpoint the problem.  This is becoming a blocking issue for me.

**Comment by alexandrebodin** [comment]
@Stefan-SW I cannot reproduce this in the latest rc.12. Can you upgrade and test it again ?

**Comment by CodeToTee** [comment]
I'm also seeing this issue on rc.12, but not in all relations...some populate, some don't.  If/when I see a pattern, I'll add those details.

**Comment by Stefan-SW** [comment]
Unfortunately the behaviour remains unchanged.

### Clarification
At the time i issued the ticket I assumed this was an issue with the REST API - maybe it is not.
I consider the behaviour erroneous/unexpected since the Admin Panel displays the `published` status on the relation and one cannot choose wehter to connect the published or draft version of the other document. 
Given the display I think it is fair to assume the relation points to the published version.

![image](https://github.com/user-attachments/assets/49b58728-e784-4897-bcb7-27255b092e0c)


### Reproducing the behaviour with rc12
Updated strapi to version 5.0.0-rc.12:

`package.json`
```
 19   │   "dependencies": {
 20 ~ │     "@strapi/plugin-cloud": "5.0.0-rc.12",
 21 ~ │     "@strapi/plugin-users-permissions": "5.0.0-rc.12",
 22 ~ │     "@strapi/strapi": "5.0.0-rc.12",
 23   │     "better-sqlite3": "9.4.3",
 24   │     "react": "^18.0.0",
 25   │     "react-dom": "^18.0.0",
 26   │     "react-router-dom": "^6.0.0",
 27   │     "styled-components": "^6.0.0"
```

`yarn.lock:`
```
...
3393   │ 
3394 ~ │ "@strapi/strapi@npm:5.0.0-rc.12":
3395 ~ │   version: 5.0.0-rc.12
3396 ~ │   resolution: "@strapi/strapi@npm:5.0.0-rc.12"
3397   │   dependencies:
3398   │     "@pmmmwh/react-refresh-webpack-plugin": "npm:0.5.11"
...
```

Added new principal and dependet documents - the result remains the same, the principal is not populated.

`❯ http 'localhost:1337/api/dependents?populate=principal'`

``` json
{
    "data": [
        {
            "Name": "the dependent entity",
            "createdAt": "2024-08-13T08:52:14.348Z",
            "documentId": "u48wv1ugkfwmj3dm41hzznz1",
            "id": 1,
            "locale": null,
            "principal": null,
            "publishedAt": "2024-08-13T08:52:14.340Z",
            "updatedAt": "2024-08-13T08:52:14.348Z"
        },
        {
            "Name": "the dependent entity - new",
            "createdAt": "2024-09-03T10:09:02.341Z",
            "documentId": "gp0cslsd926irna1yil6lmyj",
            "id": 3,
            "locale": null,
            "principal": null,
            "publishedAt": "2024-09-03T10:09:02.337Z",
            "updatedAt": "2024-09-03T10:09:02.341Z"
        }
    ],
    "meta": {
        "pagination": {
            "page": 1,
            "pageCount": 1,
            "pageSize": 25,
            "total": 2
        }
    }
}
```

The sqlite database tables suggest that the relation points to the draft version:

`principals`
```
id|document_id             |name                      |created_at   |updated_at   |published_at |created_by_id|updated_by_id|locale|
--+------------------------+--------------------------+-------------+-------------+-------------+-------------+-------------+------+
 1|knjgnlldejwhim7zqtv9wfu9|the principal entity      |1723539118788|1723539118788|             |            1|            1|      |
 2|knjgnlldejwhim7zqtv9wfu9|the principal entity      |1723539118788|1723539118788|1723539118806|             |             |      |
 5|vzsvpmtbo9fhv0gw71cvgy3y|the principal entity - new|1725358110498|1725358110498|             |            1|            1|      |
 6|vzsvpmtbo9fhv0gw71cvgy3y|the principal entity - new|1725358110498|1725358110498|1725358110503|             |             |      |
```

`dependents`
```
id|document_id             |name                      |created_at   |updated_at   |published_at |created_by_id|updated_by_id|locale|
--+------------------------+--------------------------+-------------+-------------+-------------+-------------+-------------+------+
 1|u48wv1ugkfwmj3dm41hzznz1|the dependent entity      |1723539134348|1723539134348|1723539134340|            1|            1|      |
 3|gp0cslsd926irna1yil6lmyj|the dependent entity - new|1725358142341|1725358142341|1725358142337|            1|            1|      |
```

`dependents_principal_lnk`
```
id|dependent_id|principal_id|
--+------------+------------+
 1|           1|           1|
 2|           3|           5|
```

**Comment by alexandrebodin** [comment]
From the data I'm seeing the dependents doesn't have D&P publish setup right ? I think we have someone working on this already and the pbl isn't in the API but how the relations are managed during publish/unpublish etc. 

cc @Marc-Roig

**Comment by Stefan-SW** [comment]
> From the data I'm seeing the dependents doesn't have D&P publish setup right ? 

correct, the principal has D&P enabled, the dependent has not.

**Comment by alexandrebodin** [comment]
Ok, So the behavior will be the following once fixed when you populate `principal`

- When fetching your `dependent` you will see the published version of the `principal` (this is where the db link is not created correctly right now because it's a unidirectional relation and we are working on a fix) 
- if the `principal` is not published yet you can see it with the status=draft option like you tested already

**Comment by Marc-Roig** [comment]
The underlying issue will be fixed in `5.0.0-rc.13` (will be released today), following @alexandrebodin and after the fix you should be able to get your draft & published relations from your api.

**Comment by Stefan-SW** [comment]
Sounds Great! Thank you!

**Comment by Stefan-SW** [comment]
Unfortunately the bug wasn't fixed in rc-13 and ist still present in v5.0.1.

**Comment by Stefan-SW** [comment]
In version 5.0.0-beta.0 the issue is fixed.

The `dependet` collection has D&P **disabled**
![image](https://github.com/user-attachments/assets/0d70d9ad-9834-44ee-bef6-553270b11928)

Now if I query the collection and populate the `pricipal` I get the expected result:
`❯ http http://localhost:1337/api/dependents populate==principal`
```
HTTP/1.1 200 OK

{
    "data": [
        {
            "Name": "dependent",
            "createdAt": "2024-09-26T11:29:01.296Z",
            "documentId": "ofinduqyy5b6xdr2xasdtu2i",
            "id": 1,
            "locale": null,
            "principal": {
                "createdAt": "2024-09-26T11:28:16.730Z",
                "documentId": "awwx336iajyci01af0htkeb0",
                "id": 4,
                "locale": null,
                "name": "published principal",
                "publishedAt": "2024-09-26T11:28:40.959Z",
                "updatedAt": "2024-09-26T11:28:40.945Z"
            },
            "publishedAt": "2024-10-02T08:31:52.661Z",
            "updatedAt": "2024-10-02T08:31:52.674Z"
        }
    ],
    "meta": {
        "pagination": {
            "page": 1,
            "pageCount": 1,
            "pageSize": 25,
            "total": 1
        }
    }
}
```

If a draft version of the principal is related to the dependent:
![image](https://github.com/user-attachments/assets/651f6d46-78f2-4b33-b341-51c6af846caa)

A query to the dependents endpoint won't return the principal unless the `status==draft` query parameter is set:
(without status)
`❯ http http://localhost:1337/api/dependents populate==principal`
```
HTTP/1.1 200 OK

{
    "data": [
        {
            "Name": "dependent",
            "createdAt": "2024-09-26T11:29:01.296Z",
            "documentId": "ofinduqyy5b6xdr2xasdtu2i",
            "id": 1,
            "locale": null,
            "principal": null,
            "publishedAt": "2024-10-02T08:53:02.689Z",
            "updatedAt": "2024-10-02T08:53:02.698Z"
        }
    ],
    "meta": {
        "pagination": {
            "page": 1,
            "pageCount": 1,
            "pageSize": 25,
            "total": 1
        }
    }
}```

(with status)
`❯ http http://localhost:1337/api/dependents populate==principal status==draft`
```HTTP/1.1 200 OK

{
    "data": [
        {
            "Name": "dependent",
            "createdAt": "2024-09-26T11:29:01.296Z",
            "documentId": "ofinduqyy5b6xdr2xasdtu2i",
            "id": 1,
            "locale": null,
            "principal": {
                "createdAt": "2024-09-26T11:28:30.905Z",
                "documentId": "sfk167ra0s6b8jh338ise0l1",
                "id": 3,
                "locale": null,
                "name": "draft principal",
                "publishedAt": null,
                "updatedAt": "2024-09-26T11:28:30.905Z"
            },
            "publishedAt": "2024-10-02T08:53:02.689Z",
            "updatedAt": "2024-10-02T08:53:02.698Z"
        }
    ],
    "meta": {
        "pagination": {
            "page": 1,
            "pageCount": 1,
            "pageSize": 25,
            "total": 1
        }
    }
}
```

I assume that is the intended behavior?
